src/components/pagination/__test__/Pagination.test.tsx

```diff
@@ -0,0 +1,65 @@
+import { render, screen, fireEvent, act } from '@testing-library/react';
+import { Route, Routes, MemoryRouter } from 'react-router-dom';
+import Pagination from '../Pagination';
+import '@testing-library/jest-dom';
+
+describe('Pagination component', () => {
+  const cardPerPage = 10;
+  const totalCard = 50;
+  const mockPaginate = jest.fn();
+
+  test('updates URL query parameter when page changes', async () => {
+    await act(async () => {
+      render(
+        <MemoryRouter initialEntries={['/search/page/1']}>
+          <Routes>
+            <Route
+              path="/search/page/:page"
+              element={
+                <Pagination
+                  cardPerPage={cardPerPage}
+                  totalCard={totalCard}
+                  paginate={mockPaginate}
+                />
+              }
+            />
+          </Routes>
+        </MemoryRouter>
+      );
+    });
```

Comment:
Create render setup method like in src/components/header/__test__/Header.test.tsx

Reply:
will be fixed in the next PR